src/m365/spo/commands/serviceprincipal/serviceprincipal-permissionrequest-list.ts

```diff
@@ -43,8 +45,23 @@ class SpoServicePrincipalPermissionRequestListCommand extends SpoCommand {
         throw response.ErrorInfo.ErrorMessage;
       }
       else {
+        let spoWebAppServicePrincipalPermissionRequestResult: SPOWebAppServicePrincipalPermissionRequest[] = [];
+
         const result: SPOWebAppServicePrincipalPermissionRequest[] = json[json.length - 1]._Child_Items_;
-        logger.log(result.map(r => {
+        if (result.length > 0) {
+          const spoClientExtensibilityWebApplicationPrincipalId = await this.getSPOClientExtensibilityWebApplicationPrincipalId();
+          if (spoClientExtensibilityWebApplicationPrincipalId !== null) {
+            const oAuth2PermissionGrants: string[] | null = await this.getOAuth2PermissionGrants(spoClientExtensibilityWebApplicationPrincipalId);
+            if (oAuth2PermissionGrants) {
+              spoWebAppServicePrincipalPermissionRequestResult = result.filter(x => oAuth2PermissionGrants!.indexOf(x.Scope) === -1);
```

Comment:
Since we've just checked that `oAuth2PermissionGrants` we don't need to use `oAuth2PermissionGrants!` later on. `oAuth2PermissionGrants` is sufficient.